src/metabase/api/public.clj

```diff
@@ -97,6 +94,19 @@
   (fn [metadata final-metadata context]
     (orig-reducedf metadata (transform-results final-metadata) context)))
 
+(defn- run-query-for-card-with-id-async-run-fn
+  "Create the `:run` function used for [[run-query-for-card-with-id-async]] and [[public-dashcard-results-async]]."
+  [qp-runner export-format]
+  (fn [query info]
+    (qp.streaming/streaming-response
+        [{:keys [reducedf], :as context} export-format (u/slugify (:card-name info))]
+        (let [context  (assoc context :reducedf (public-reducedf reducedf))
+              in-chan  (binding [api/*current-user-permissions-set* (atom #{"/"})]
```

Comment:
this binding makes me nervous. Does that permission set get conveyed onto the thread that does the work and populates the resulting channel? I'm guessing you are sure but just want to highlight it.

Reply:
yeah, the binding does get picked up. It is a little bit of an awkward way of setting the perms but I'm not sure what a better way of doing it would be. Worst case scenario it would just fail closed.